core/src/main/java/hudson/util/io/ZipArchiver.java

```diff
@@ -48,12 +53,21 @@
     private final byte[] buf = new byte[8192];
     private final ZipOutputStream zip;
     private final OpenOption[] openOptions;
+    private final String prefix;
 
     ZipArchiver(OutputStream out) {
-        this(out, false);
+        this(out, false, "");
     }
 
-    ZipArchiver(OutputStream out, boolean failOnSymLink) {
+    // Restriction added for clarity, it's a package class, you should not use it outside of Jenkins core
+    @Restricted(NoExternalUse.class)
+    ZipArchiver(OutputStream out, boolean failOnSymLink, String prefix) {
+        if (StringUtils.isBlank(prefix)) {
+            this.prefix = "";
+        } else {
+            this.prefix = Util.ensureEndsWith(prefix.trim(), "/");
```

Comment:
This isn't user input so I expect that trimming will make the code fail when a file/directory name legitimately starts/end with whitespace.

Reply:
FTR the prefix is "just" used inside the Zip, not to retrieve the real file, but for the outcome inside the ZipEntry. If you put as prefix "test", it will work, your Zip will just contain a parent folder being "test".

OTOH If you have a real baseFile with a space at the end, your zip will be "cleaner" but if you expect that space to be there, yeah, your script could be not working.

**No strong opinion**. The trim version is not equivalent to the version before the security fix.

@oleg-nenashev @daniel-beck , please convince yourselves ;)

Reply:
No strong opinion either, whatever works

Reply:
There seems to be no reason to do this, only a reason not to, as this isn't users copying text from Outlook into build parameters. At a minimum, it seems to be beyond the scope of this change as an unrelated RFE (unless demonstrated to have behaved this way in 2.274).

Reply:
I would prefer to revert the trim and then ship it so we get it into 2.276

Reply:
Resolved, thanks for the discussion there!